alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetMonitorAction.kt

```diff
@@ -140,40 +130,72 @@ class TransportGetMonitorAction @Inject constructor(
         }
     }
 
-    private fun getAssociatedWorkflows(id: String): List<AssociatedWorkflow> {
-        try {
-            val associatedWorkflows = mutableListOf<AssociatedWorkflow>()
-            val queryBuilder = QueryBuilders.nestedQuery(
-                WORKFLOW_DELEGATE_PATH,
-                QueryBuilders.boolQuery().must(
-                    QueryBuilders.matchQuery(
-                        WORKFLOW_MONITOR_PATH,
-                        id
-                    )
-                ),
-                ScoreMode.None
-            )
-            val searchRequest = SearchRequest()
-                .indices(ScheduledJob.SCHEDULED_JOBS_INDEX)
-                .source(SearchSourceBuilder().query(queryBuilder).fetchField("_id"))
-            val response: SearchResponse = client.execute(SearchAction.INSTANCE, searchRequest).get()
-
-            for (hit in response.hits) {
-                XContentType.JSON.xContent().createParser(
-                    xContentRegistry,
-                    LoggingDeprecationHandler.INSTANCE,
-                    hit.sourceAsString
-                ).use { hitsParser ->
-                    val workflow = ScheduledJob.parse(hitsParser, hit.id, hit.version)
-                    if (workflow is Workflow) {
-                        associatedWorkflows.add(AssociatedWorkflow(hit.id, workflow.name))
+    private fun getAssociatedWorkflows(
+        id: String,
+        getResponse: GetResponse,
+        monitor: Monitor?,
+        actionListener: ActionListener<GetMonitorResponse>
+    ) {
+        val queryBuilder = QueryBuilders.nestedQuery(
+            WORKFLOW_DELEGATE_PATH,
+            QueryBuilders.boolQuery().must(
+                QueryBuilders.matchQuery(
+                    WORKFLOW_MONITOR_PATH,
+                    id
+                )
+            ),
+            ScoreMode.None
+        )
+        val searchRequest = SearchRequest()
+            .indices(ScheduledJob.SCHEDULED_JOBS_INDEX)
+            .source(SearchSourceBuilder().query(queryBuilder).fetchField("_id"))
+        client.execute(
```

Comment:
thanks for this find!

can we do this in a try catch with client.suspendUntil()